### ISSUE TITLE
Custom Emojis with Unicode Equivalents

### ISSUE THREAD

**Issue by jeremyredhead** [Garmelon/euphoxide#1]
So there are a couple custom emojis that actually have, or arguably have, Unicode equivalents now.

First up is `:orange_heart:` / https://emojipedia.org/orange-heart/ 
- The codepoint was added to Unicode in 2017, long after heim development had stalled.
- It looks the same most anywhere.
- Every platform uses the same shortcode.
- Conclusion: I see no reason not to add the codepoint to the emoji.txt file, and would heartily endorse doing so.

Second is `:spider:` / https://emojipedia.org/spider/
- The codepoint was added in 2014 (as a black & white text thingy) and added to Emoji 1.0 in 2015. Not sure why neither emoji-annotation-to-unicode nor heim was ever updated to use the official code point (maybe b/c it requires variation selector-16 to show as an emoji?). Need to dig into the history to figure out if this was an oversight or Intentional.
- Euphoria's `:spider:` is a brown spider facing downwards, whilst platform emojis vary but are typically black, and often facing upwards.
- The shortcode is, again, the same everywhere. 
- Conclusion: I don't know if I'd recommend adding it to the file, but I think it wouldn't hurt. Not sure if the plain codepoint or the emoji variant should be added tho

Last is `:bot:` / https://emojipedia.org/robot/ 
- The codepoint was added to Unicode 8.0 in 2015 (as "Robot Face") and Emoji 1.0. The euphoria emoji was designed by jedevc IIRC, and was added because... because why not? I forget, I need to go dig up that history
- Euphoria's `:bot:` has a wide grin, green eyes, a jovial expression, absolutely adorable, 10/10. Platform emojis tend to look like the archetypal 50's sci-fi robot, and very soulless. Rather different in connotation, even if nominally the same in denotation. 
- The shortcode for Unicode Robot is either `:robot:` or `:robot_face:`. No harmony.
- Conclusion: It could be argued they're Close Enough but I would recommend **against** mapping `:bot:` to 🤖. Because euphoria's `:bot:` is so much more adorable and clearly superior.

**Comment by Garmelon** [comment]
Thanks for the detailed overview.

> maybe b/c it requires variation selector-16 to show as an emoji?

I wonder how well the variation selector and terminal emulators play together

> Conclusion: It could be argued they're Close Enough but I would recommend against mapping :bot: to robot. Because euphoria's :bot: is so much more adorable and clearly superior.

The alternative is not euphoria's bot emoji, but the literal text `:bot:`. :robot: is pretty close, though not looking as happy.

**Comment by jeremyredhead** [comment]
> I wonder how well the variation selector and terminal emulators play together

Yeah, me too. Guess someone needs to test that? Or maybe, I hope, someone else in the world has previously wondered this and already compiled an emoji in terminal emulators support/compatibility table.

How many of the currently added emojis/codepoints in emoji.txt normally require var-16 to show in emoji & not text form?

> The alternative is not euphoria's bot emoji, but the literal text `:bot:`. 

Ye, I know... Though one might semi-humorously & absurdly suggest mapping the custom emoji without so-called ""standardized"" equivalents to either PUA codepoints (but eugh), or ZWJ sequences, e.g. `:greenduck:` could be mapped to, IDFK, 1f7e9 200d 1f986, i.e. 🟩‍🦆. Then the user could simply add glyphs for these sequences to a font or font config or what have you. This is, of course, an entirely serious suggestion ;)

> 🤖 is pretty close, though not looking as happy.

Yea, which is why I didn't suggest mapping `:bot:` to it. It makes me sad :(
